collab-database/src/workspace_database/body.rs

```diff
@@ -56,7 +56,7 @@ impl WorkspaceDatabase {
   /// Create a new [DatabaseMeta] for the given database id and view id
   /// use [Self::update_database] to attach more views to the existing database.
   ///
-  pub fn add_database(&mut self, database_id: &str, view_ids: Vec<String>) {
+  pub fn add_database(&mut self, database_id: &str, view_ids: Vec<String>) -> TransactionMut {
```

Comment:
It's better to move `add_database` logic to body and provide transaction from outside. What if you want to call this method, while you already had transaction acquired? This would cause a deadlock.